edx_api/integration_test.py

```diff
@@ -4,41 +4,85 @@
 Thes run against actual edx apis.
 """
 import os
+
 import pytest
+from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
+
 from .client import EdxApi
 
 
 ACCESS_TOKEN = os.getenv('ACCESS_TOKEN')
 BASE_URL = os.getenv('BASE_URL', 'http://localhost:8000/')
+COURSE_ID = os.getenv('COURSE_ID', 'course-v1:edX+DemoX+Demo_Course')
 
 # pylint: disable=invalid-name
 require_integration_settings = pytest.mark.skipif(
-    not ACCESS_TOKEN or not BASE_URL,
+    not ACCESS_TOKEN or not BASE_URL or not COURSE_ID,
     reason="You must specify the credential env vars"
 )
 
 
+def get_username():
+    """
+    Helper function to retrieve the username given the current access token.
+    Needed until this library does not implement a proper client
+    """
+    api = EdxApi({'access_token': ACCESS_TOKEN}, base_url=BASE_URL)
+    requester = api.get_requester()
+    resp = requester.get(urljoin(BASE_URL, "/api/mobile/v0.5/my_user_info"))
+    data = resp.json()
+    return data['username']
+
+
 @require_integration_settings
 def test_course_structure():
     """
     Pull down the course structure and validate it has the correct entries.
+    This test assumes that the used user can access the course.
     """
     api = EdxApi({'access_token': ACCESS_TOKEN}, base_url=BASE_URL)
-    structure = api.course_structure.course_blocks(
-        'course-v1:edX+DemoX+Demo_Course',
-        'staff'
-    )
+    username = get_username()
+    structure = api.course_structure.course_blocks(COURSE_ID, username)
 
     titles = [
         block.title for block in structure.root.children
         if block.visible
     ]
 
-    assert titles == [
-        'Introduction',
-        'Example Week 1: Getting Started',
-        'Example Week 2: Get Interactive',
-        'Example Week 3: Be Social',
-        'About Exams and Certificates',
-        'holding section',
+    assert len(titles) > 0
+
+    if COURSE_ID == 'course-v1:edX+DemoX+Demo_Course':
```

Comment:
This seems like a much weaker assertion to me. I'd like to figure out how we can structure this so we don't have conditionals like this in our tests. This could be passing with a different course and not returning the correct results.

Any thoughts?